Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -171,59 +368,139 @@ - (void)selectSubcategoryDone:(OAPOICategory *)category keys:(NSMutableSet<NSStr
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    OAMenuSimpleCell* cell = [tableView dequeueReusableCellWithIdentifier:kCellTypeTitleDescCollapse];
-    if (cell == nil)
-    {
-        NSArray *nib = [[NSBundle mainBundle] loadNibNamed:kCellTypeTitleDescCollapse owner:self options:nil];
-        cell = (OAMenuSimpleCell *) nib[0];
-        cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+    if (!_searchMode) {
+        OAMenuSimpleCell *cell = [tableView dequeueReusableCellWithIdentifier:kCellTypeTitleDescCollapse];
+        if (cell == nil) {
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:kCellTypeTitleDescCollapse owner:self options:nil];
+            cell = (OAMenuSimpleCell *) nib[0];
+        }
+
+        if (cell) {
+            OAPOICategory *category = _categories[indexPath.row];
+            NSSet<NSString *> *subtypes = [_acceptedTypes objectForKey:category];
+            NSInteger countAcceptedTypes = subtypes.count;
+            NSInteger countAllTypes = category.poiTypes.count;
+            BOOL isSelected = [_filter isTypeAccepted:category] && subtypes.count > 0;
+            cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+
+            cell.contentView.backgroundColor = [UIColor whiteColor];
+            cell.separatorInset = UIEdgeInsetsMake(0.0, 66.0, 0.0, 0.0);
+
+            cell.textView.text = category.nameLocalized;
+            cell.textView.textColor = [UIColor blackColor];
+
+            UIImage *img = [[category icon] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+            cell.imgView.image = img;
+            cell.imgView.tintColor = isSelected ? UIColorFromRGB(color_primary_purple) : UIColorFromRGB(color_tint_gray);
+            cell.imgView.contentMode = UIViewContentModeCenter;
+
+            NSString *descText;
+            if (subtypes == [OAPOIBaseType nullSet] || countAllTypes == countAcceptedTypes)
+                descText = [NSString stringWithFormat:@"%@ - %lu", OALocalizedString(@"shared_string_all"), countAllTypes];
+            else
+                descText = [NSString stringWithFormat:@"%lu/%lu", countAcceptedTypes, countAllTypes];
+            cell.descriptionView.hidden = false;
+            cell.descriptionView.text = descText;
+            cell.descriptionView.textColor = UIColorFromRGB(color_text_footer);
+
+            [cell updateConstraints];
+            return cell;
+        }
+    } else {
+        OAMenuSimpleCell *cell = [tableView dequeueReusableCellWithIdentifier:kCellTypeTitle];
```

Comment:
Confusing code, why do you create the same cell twice? You should create it once and only then adjust the data according to the _searchMode flag.